src/Tasks/ManifestUtil/SecurityUtil.cs

```diff
@@ -572,7 +572,11 @@ public static void SignFile(string certPath, SecureString certPassword, Uri time
         private static bool UseSha256Algorithm(X509Certificate2 cert)
         {
             Oid oid = cert.SignatureAlgorithm;
-            return string.Equals(oid.FriendlyName, "sha256RSA", StringComparison.OrdinalIgnoreCase);
+            // Issue 6732: Clickonce does support sha384/sha512 hash so we default to sha256 
```

Comment:
```suggestion
            // Issue 6732: Clickonce does not support sha384/sha512 hash so we default to sha256 
```

?

Reply:
I'm not sure this question makes sense but: Is it possible for a cert to have sha384/sha512 and _not_ have 256?

Reply:
We use the signature algorithm of the cert to decide what file digest algorithm to use for the clickonce manifest files. So for certs with sha1 signature, we choose sha1 file digest algorithm for clickonce manfest. With this change, we will use sha256 file digest algorithm for clickonce manifest if the signature algorithm in the cert is sha256 or sha384 or sha512.

The cert's signature algorithm is the algorithm used to create the signature of the cert and can be only 1 value like sha1/sha256/sha384/sha512.

Reply:
Ah, so it's an independent signature/algorithm but it makes sense to use the strongest possible one based on the signal of the cert provided?

Reply:
Correct.